### PR TITLE
FAT-9210 Develop ansible plugin to add fusion service into HDP using …

### PR DESCRIPTION
**Pull request by Lozmanov**
…mpacks

### CODE REVIEW

File: mpack/ambari_mpack.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+        self.api_client.post(component_url, content_type="application/json")
+        self.changed = True
+
+    def _create_service(self):
+        self.api_client.post(self.api_client.clusters(self.cluster_name).services(self.service_name).url,
+                             content_type="application/json")
+        self._update()
+        self.changed = True
+
+    def _restart_stale_config_components(self):
```
Comment (nikitamishustin) [inline mpack/ambari_mpack.py:213]: This seems almost the same, why do we need to restart stale services twice?

File: mpack/ambari_mpack.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    def deleted(self):
+        if self.status != "ABSENT":
+            self._stop_service()
+            self._delete_service()
+
+    def _stop_service(self):
+        try:
+            self.api_client.clusters(self.cluster_name).services(self.service_name).stop().wait()
+            self.changed = True
+        except:
```
Comment (nikitamishustin) [inline mpack/ambari_mpack.py:236]: Using a bare except is not recommended.
"except Exception", at least

File: mpack/ambari_mpack.py [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+    cluster_name = params['cluster_name']
+
+    api_client = Ambari(api_host, port=api_port, username=api_username, password=api_password)
+
+    if params["service"] is not None:
+        components = params['components']
+        config_path = params['config_path']
+        component_hosts_count = params['count']
+        extra_config = None
+        if 'extra' in params:
```
Comment (nikitamishustin) [inline mpack/ambari_mpack.py:283]: "extra" always present in the "params"
This check will always be "True"

Comment (Lozmanov) [reply mpack/ambari_mpack.py:283]: I'll change it to unnecessary extra in params in future. Actually, you can run FUSION_CLIENT isntallation without any extra params

File: mpack/ambari_mpack.py (hunk 2 above)
Comment (Lozmanov) [reply mpack/ambari_mpack.py]: done

File: mpack/ambari_mpack.py (hunk 1 above)
Comment (Lozmanov) [reply mpack/ambari_mpack.py:219]: fixed

File: mpack/ambari_mpack.py (hunk 3 above)
Comment (nikitamishustin) [reply mpack/ambari_mpack.py:289]: Yeah, but "extra" and other stuff anyway is in the "params"
Like this:
```
{
    "cluster_name": "OSUSL-01",
    "components": null,
    "config_path": null,
    "count": null,
    "extra": null,
    "host": "localhost",
    "login": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
    "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
    "port": "8080",
    "service": null,
    "state": "present"
}
```
So this check can look like this:
```
if params['extra'] is not None:
```

**Pull request merged**